platform/linux-generic/include/odp_schedule_if.h

```diff
@@ -42,40 +44,50 @@ typedef struct {
+typedef void *queue_t;
+
+typedef	int (*queue_init_global_fn_t)(void);
+typedef int (*queue_term_global_fn_t)(void);
+typedef queue_t (*queue_from_ext_fn_t)(odp_queue_t handle);
+typedef odp_queue_t (*queue_to_ext_fn_t)(queue_t handle);
+typedef int (*queue_enq_fn_t)(queue_t handle, odp_buffer_hdr_t *);
+typedef int (*queue_enq_multi_fn_t)(queue_t handle, odp_buffer_hdr_t **, int);
+typedef	odp_buffer_hdr_t *(*queue_deq_fn_t)(queue_t handle);
+typedef	int (*queue_deq_multi_fn_t)(queue_t handle, odp_buffer_hdr_t **, int);
+typedef odp_pktout_queue_t (*queue_get_pktout_fn_t)(queue_t handle);
+typedef void (*queue_set_pktout_fn_t)(queue_t handle, odp_pktio_t pktio,
+				      int index);
+typedef odp_pktin_queue_t (*queue_get_pktin_fn_t)(queue_t handle);
+typedef void (*queue_set_pktin_fn_t)(queue_t handle, odp_pktio_t pktio,
+				     int index);
+typedef void (*queue_set_enq_fn_t)(queue_t handle, queue_enq_fn_t func);
+typedef void (*queue_set_enq_multi_fn_t)(queue_t handle,
+					 queue_enq_multi_fn_t func);
+typedef void (*queue_set_deq_fn_t)(queue_t handle, queue_deq_fn_t func);
+typedef void (*queue_set_deq_multi_fn_t)(queue_t handle,
+					 queue_deq_multi_fn_t func);
+typedef void (*queue_set_type_fn_t)(queue_t handle, odp_queue_type_t type);
 
 /* Queue functions towards other internal components */
 typedef struct {
-	int (*queue_init_global)(void);
-	int (*queue_term_global)(void);
-	int (*queue_enq)(queue_entry_t *queue, odp_buffer_hdr_t *buf_hdr);
-	int (*queue_enq_multi)(queue_entry_t *queue,
-			       odp_buffer_hdr_t *buf_hdr[], int num);
-	odp_buffer_hdr_t *(*queue_deq)(queue_entry_t *queue);
-	int (*queue_deq_multi)(queue_entry_t *queue,
-			       odp_buffer_hdr_t *buf_hdr[], int num);
-	odp_pktout_queue_t (*queue_get_pktout)(queue_entry_t *queue);
-	void (*queue_set_pktout)(queue_entry_t *queue, odp_pktio_t pktio,
-				 int index);
-	odp_pktin_queue_t (*queue_get_pktin)(queue_entry_t *queue);
-	void (*queue_set_pktin)(queue_entry_t *queue, odp_pktio_t pktio,
-				int index);
-	void (*queue_set_enq_func)(queue_entry_t *queue, enq_func_t func);
-	void (*queue_set_enq_multi_func)(queue_entry_t *queue,
-					 enq_multi_func_t func);
-	void (*queue_set_deq_func)(queue_entry_t *queue, deq_func_t func);
-	void (*queue_set_deq_multi_func)(queue_entry_t *queue,
-					 deq_multi_func_t func);
-	void (*queue_set_type)(queue_entry_t *queue, odp_queue_type_t type);
+	queue_init_global_fn_t init_global;
+	queue_term_global_fn_t term_global;
+	queue_from_ext_fn_t from_ext;
+	queue_to_ext_fn_t to_ext;
+	queue_enq_fn_t enq;
+	queue_enq_multi_fn_t enq_multi;
+	queue_deq_fn_t deq;
+	queue_deq_multi_fn_t deq_multi;
+	queue_get_pktout_fn_t get_pktout;
+	queue_set_pktout_fn_t set_pktout;
+	queue_get_pktin_fn_t get_pktin;
+	queue_set_pktin_fn_t set_pktin;
+	queue_set_enq_fn_t set_enq_fn;
+	queue_set_enq_multi_fn_t set_enq_multi_fn;
+	queue_set_deq_fn_t set_deq_fn;
+	queue_set_deq_multi_fn_t set_deq_multi_fn;
+	queue_set_type_fn_t set_type;
```

Comment:
So, could you please drop the `queue_` prefix from the beginning (in the previous patch)?

Also, could you please point, how other components will use function types? Will they contain additional structures with these functions? Or just implementations of this interface. In the latter case I'd prefer to live without additional types.